Update scenario management overview from develop (#571)
Avaiga/taipy-doc: docs/tutorials/complete_applications/scenario_management_overview/step_04/step_04.md

> You can download the code for
<a href="./../src/step_04.py" download>Step 4</a> 
or all the steps <a href="./../src/src.zip" download>here</a>. 

# Cycles

*Estimated Time for Completion: 15 minutes; Difficulty Level: Intermediate*

[Cycles](../../../../manuals/core/concepts/cycle.md) have been introduced to reflect business situations our customers frequently encounter and can be extremely useful. 

> For instance, a large fast food chain wants to generate sales forecasts for its stores every week. When creating a given scenario, it needs to be attached to a given week. Typically, only one scenario will be shared among all the scenarios made for a specific week. This specific 'official' scenario is known as the 'Primary' scenario in Taipy.

Reviewer: ```suggestion
For example, a big fast-food chain wants to predict its store sales every week. When they create a scenario 
for a particular week, it has to be linked to that specific week. Usually, there will be just one scenario 
used for all the scenarios created for a particular week. 
This special 'official' scenario is called the 'Primary' scenario in Taipy.
```

Note that Cycles can be ignored entirely if the business problem has no time frequency. 

> In this step, scenarios are attached to a MONTHLY cycle. Using Cycles, the developer benefits from specific Taipy's functions to navigate through these Cycles. For instance, by providing the Cycle, Taipy can get all the scenarios created in a month. You can also easily get every primary scenario generated for the past X months to monitor KPIs over time.

Reviewer: ```suggestion
In this step, scenarios are connected to a MONTHLY cycle. With the help of Cycles, developers can take advantage 
of Taipy's specific functions to work with these Cycles. For instance, by providing the Cycle, Taipy can retrieve 
all the scenarios created within a month. You can also easily access every primary scenario generated 
in the past X months to track KPIs over time.
```

> Let’s slightly change the filter function by passing the month as an argument to get started. You must create a new Data Node representing the month (see the steps below).

Reviewer: ```suggestion
Let's make a slight adjustment to the filter function by passing the month as an argument. You'll need to create a new Data Node that represents the month. Follow the steps below:
```

        ...

        scenario_cfg = Config.configure_scenario(id="my_scenario",
                                                            task_configs=[task_filter_cfg,
                                                                          task_count_values_cfg],
                                                            frequency=Frequency.MONTHLY)

        ```

> As you can see, a Cycle is activated once you have set the desired frequency of the scenario. In this code snippet, since we have specified `frequency=Frequency.MONTHLY`, the corresponding scenario is automatically attached to the correct period (month) when it is created. 

Reviewer: ```suggestion
In the code snippet, you can observe that a Cycle becomes active when you've set the desired frequency for the scenario. In this code snippet, since we have specified `frequency=Frequency.MONTHLY`, the corresponding scenario is automatically attached to the correct period (month) when it is created. 
```

                                name="Scenario 2022/10/5")
```

> Scenario 1 and Scenario 2 are two scenario entities/instances created from the same scenario configuration. They belong to the same Cycle but don't share the same Data Nodes. By default, each scenario instance has its own data node instances. They are not shared with any other scenario. The Scope concept can modify this behavior, which is covered in the next step.

Reviewer: ```suggestion
Scenario 1 and Scenario 2 are two separate scenario entities created using the same scenario configuration. 
They are part of the same Cycle but have different Data Nodes. By default, each scenario instance 
has its own data node instances, and they are not shared with any other scenario. 
The Scope concept can change this behavior, and we'll discuss it in the next step.
```

```
Month Data Node of Scenario 1 10
Month Data Node of Scenario 2 10
[2022-12-22 16:20:04,746][Taipy][INFO] job JOB_filter_by_month_... is completed.
[2022-12-22 16:20:04,833][Taipy][INFO] job JOB_count_values_... is completed.
[2022-12-22 16:20:05,026][Taipy][INFO] job JOB_filter_by_month_... is completed.
[2022-12-22 16:20:05,084][Taipy][INFO] job JOB_count_values_... is completed.
[2022-12-22 16:20:05,317][Taipy][INFO] job JOB_filter_by_month_... is completed.
[2022-12-22 16:20:05,376][Taipy][INFO] job JOB_count_values_... is completed.
```

## Primary scenarios

In each Cycle, there is a primary scenario. A primary scenario is interesting because it represents the important scenario of the Cycle, the reference. By default, the first scenario created for a cycle is primary.

### GUI-Core visual elements

> GUI-Core elements may be used to manipulate Cycles. Cycles are visible in *scenario_selector* or *data_node_selector*. It is also possible to make a scenario primary directly through the `scenario` visual element.

Reviewer: ```suggestion
You can utilize GUI-Core elements to control Cycles. Cycles can be seen in either the *scenario_selector* or *data_node_selector*. 
Additionally, it's possible to designate a scenario as primary directly through the `scenario` visual element.
```
